src/js/select2/selection/search.js

```diff
@@ -100,6 +100,7 @@ define([
   };
 
   Search.prototype.update = function (decorated, data) {
+    var had_focus = this.$search.is(":focus");
```

Comment:
This should be `hadFocus` instead of `had_focus`, trying to avoid inconsistent styles.